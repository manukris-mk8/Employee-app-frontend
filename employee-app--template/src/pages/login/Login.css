
.outer-class{
  display: flex;
  height: 100vh;
  /* font-family:  'Poppins' ; */
    font-family: 'Nunito Sans';
}


