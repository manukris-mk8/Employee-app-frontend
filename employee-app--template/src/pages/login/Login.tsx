import { LeftContainer } from "../../containers/left-container/LeftContainer"
import { RightContainer } from "../../containers/right-container/RightContainer"
// import { UncontrolledLogin } from "../../containers/right-container/UncontrolledLogin"
import './Login.css'

const Login = () => {
    return(
        <div>
            <div className='outer-class'>
            <LeftContainer />
            <RightContainer />
            {/* <UncontrolledLogin/> */}
        </div>
        </div>
    )
}

export default Login;