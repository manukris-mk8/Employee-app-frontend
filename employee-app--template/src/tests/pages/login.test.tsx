import { describe, it, expect, beforeEach, vi } from "vitest";
import { render, screen, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import userEvent from "@testing-library/user-event";
import Login from "../../pages/login/Login";
// Mock react-router-dom's useNavigate
const mockNavigate = vi.fn();
vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockNavigate,
  };
});
// Mock the store
const mockStore = configureStore({
  reducer: {
    employee: (state = { employees: [] }) => state,
    department: (state = { departments: [] }) => state,
    baseApi: (state = {}) => state,
  },
});
// Mock the login mutation
const mockLoginMutation = vi.fn();
vi.mock("../../api-services/auth/login.api", () => ({
  useLoginMutation: () => [
    () => ({
      unwrap: () => mockLoginMutation(),
    }),
    { isLoading: false },
  ],
}));
describe("Login Page", () => {
  const renderLogin = () => {
    return render(
      <Provider store={mockStore}>
        <Login />
      </Provider>
    );
  };
  beforeEach(() => {
    vi.clearAllMocks();
  });
  it("should match snapshot", () => {
    const { container } = renderLogin();
    expect(container).toMatchSnapshot();
  });

  // Test 2: Render login form with all required elements
  it("should render login form with all required elements", () => {
    renderLogin();

    // Check if username input is present
    const usernameInput = screen.getByLabelText("email");
    expect(usernameInput).toBeInTheDocument();

    // Check if password input is present
    const passwordInput = screen.getByLabelText("password");
    expect(passwordInput).toBeInTheDocument();

    // Check if submit button is present
    const submitButton = screen.getByRole("button", { name: /log in/i });
    expect(submitButton).toBeInTheDocument();
    expect(submitButton).toHaveAttribute("type", "submit");
  });

  it("should show error when username is too long", async () => {
    renderLogin();
    const usernameInput = screen.getByLabelText("email");
    // Type a username longer than 30 characters
    await userEvent.type(usernameInput, "a".repeat(5));
    // Check if error message is displayed
    expect(
      screen.getByText("Invalid email id !")
    ).toBeInTheDocument();
  });


it("should handle successful login", async () => {
    mockLoginMutation.mockResolvedValueOnce({ accessToken: {token: "fake-token", id: 'fakeId',name: 'fake', email:'fake@email.com',role:'fake-role' }});

    renderLogin();

    // Fill in the form
    const usernameInput = screen.getByLabelText("email");
    const passwordInput = screen.getByLabelText("password");
    const submitButton = screen.getByRole("button", { name: /log in/i });

    await userEvent.type(usernameInput, "test@example.com");
    await userEvent.type(passwordInput, "password123");

    // Submit the form
    await userEvent.click(submitButton);

    // Wait for the login mutation to resolve and navigation to occur
    await waitFor(() => {
      expect(mockLoginMutation).toHaveBeenCalled();
      expect(mockNavigate).toHaveBeenCalledWith("/employees");
    });
  });

});