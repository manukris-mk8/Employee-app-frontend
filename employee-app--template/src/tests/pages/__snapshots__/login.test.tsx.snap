// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Page > should match snapshot 1`] = `
<div>
  <div>
    <div
      class="outer-class"
    >
      <div
        class="left-outer-class"
      >
        <div
          class="circle"
        >
          <img
            alt="kv-login"
            src="/src/assets/kv-login.jpeg"
          />
        </div>
      </div>
      <div
        class="right-outer-class"
      >
        <form
          class="login-form"
        >
          <img
            alt="kv-logo"
            src="/src/assets/kv-logo.png"
          />
          <div
            class="input-container"
          >
            <label
              for="username"
            >
              email
            </label>
            <input
              class="input-container"
              id="username"
              placeholder=" "
              required=""
              type="text"
              value=""
            />
            <button
              class="cancel-btn"
              disabled=""
              type="button"
            >
              X
               
            </button>
          </div>
          <div
            class="input-container"
          >
            <label
              for="password"
            >
              password
            </label>
            <input
              class="input-container"
              id="password"
              placeholder=" "
              required=""
              type="password"
              value=""
            />
            <button
              class="cancel-btn"
              disabled=""
              type="button"
            >
              X
               
            </button>
          </div>
          <div
            class="checkbox-container"
          >
            <input
              id="showPassword"
              type="checkbox"
            />
            <label
              for="showPassword"
            >
              Show password
            </label>
          </div>
          <button
            class="button-class"
            type="submit"
          >
            Log in
             
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
